specs/phase1/shard-transition.md

```diff
@@ -242,15 +249,17 @@ def get_shard_state_transition_result(
     beacon_state: BeaconState,
     shard: Shard,
     shard_blocks: Sequence[SignedShardBlock],
+    on_time_slot: Slot,
     validate_signature: bool=True,
 ) -> Tuple[Sequence[SignedShardBlock], Sequence[ShardState], Sequence[Root]]:
     proposals = []
     shard_states = []
     shard_state = beacon_state.shard_states[shard]
-    for slot in get_offset_slots(beacon_state, shard):
+    offset_slots = compute_offset_slots(get_latest_slot_for_shard(beacon_state, shard), on_time_slot)
```

Comment:
I think I prefer this without `on_time_slot` as in #1704 but we can discuss more if you want.

My main argument is that clients are already very used to transition states up to the slot in question so the slot information can and should already be embedded in `beacon_state`